server/controllers/CommentController.js

```diff
@@ -0,0 +1,145 @@
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async addComment(req, res) {
+    try {
+      const token = req.headers.authorization.split(' ')[1];
+      const { userId } = TokenManager.verify(token);
+      const { content } = req.body;
+      const { articleId } = req.params;
+      if (!content) {
+        return response(res, 400, 'failure', 'Enter a comment', null, null);
+      }
+      await Article.findAndCountAll({
+        where: {
+          id: articleId
+        }
+      });
+      const commentCreated = await Comment.create({
+        content,
+        userId,
+        articleId
+      });
+      if (commentCreated) {
+        return response(res, 201, 'success', 'Comment created', null, null);
+      }
+    } catch (error) {
+      return response(res, 404, 'failure', 'Article with the id not found', null, null);
+    }
+  }
+
+  /**
+   *
+   * @description Retrive all comment for an article
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async getComments(req, res) {
+    try {
+      const { articleId } = req.params;
+      const commentsQuery = await Comment.findAll({
+        where: {
+          articleId
+        }
+      });
+      const comments = [];
+      commentsQuery.map(comment => comments.push(comment.dataValues));
+      return response(res, 200, 'success', 'Comment found', null, comments);
+    } catch (error) {
+      return response(res, 404, 'failure', 'Comment not found for article id', null, null);
+    }
+  }
+
+  /**
+   *
+   * @description Update a comment
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async updateComment(req, res) {
+    try {
+      const token = req.headers.authorization.split(' ')[1];
+      const { userId } = TokenManager.verify(token);
+      const { articleId, commentId } = req.params;
+      const { content } = req.body;
+      const getCommentUpdate = await Comment.findOne({
+        where: {
+          id: commentId
+        }
+      });
+      if (getCommentUpdate.dataValues.articleId !== articleId) {
+        return response(res, 404, 'failure', 'Comment not found for article id', null, null);
+      }
+      if (getCommentUpdate.dataValues.userId !== userId) {
+        return response(res, 401, 'failure', 'You are not allowed to update another user\'s comment', null, null);
+      }
+      const updateComment = await getCommentUpdate.update({
+        content
+      });
+      if (updateComment) {
+        return response(res, 200, 'success', 'Comment updated', null, null);
+      }
+    } catch (error) {
+      return response(res, 404, 'failure', 'Comment with the id not found', null, null);
```

Comment:
I would really love to know why you are handling 404 here.

Reply:
if a query fails it throw an error automatically dat why 404 is in the catch block for example invalid articleId

Reply:
But will also account for server error thanks